opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -0,0 +1,211 @@
+/*
+  Copyright 2019 Norce.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef OPM_WELLGROUPHELPERS_HEADER_INCLUDED
+#define OPM_WELLGROUPHELPERS_HEADER_INCLUDED
+
+#include <vector>
+
+namespace Opm {
+
+
+    namespace wellGroupHelpers
+    {
+
+    inline void setCmodeGroup(const Group2& group, const Schedule& schedule, const SummaryState& summaryState, const int reportStepIdx, WellStateFullyImplicitBlackoil& wellState) {
+
+        for (const std::string& groupName : group.groups()) {
+            setCmodeGroup( schedule.getGroup2(groupName, reportStepIdx), schedule, summaryState, reportStepIdx, wellState);
+        }
+
+        if (group.isInjectionGroup()) {
+            const auto controls = group.injectionControls(summaryState);
+            wellState.setCurrentInjectionGroupControl(group.name(), GroupInjection::ControlEnum2String(controls.cmode));
+        }
+        if (group.isProductionGroup()) {
+            const auto controls = group.productionControls(summaryState);
+            wellState.setCurrentProductionGroupControl(group.name(), GroupProduction::ControlEnum2String(controls.cmode));
+        }
+    }
+
+
+    inline void accumulateGroupEfficiencyFactor(const Group2& group, const Schedule& schedule, const int reportStepIdx, double& factor) {
+        factor *= group.getGroupEfficiencyFactor();
+        if (group.parent() != "FIELD")
+            accumulateGroupEfficiencyFactor(schedule.getGroup2(group.parent(), reportStepIdx), schedule, reportStepIdx, factor);
+    }
+
+
+    inline void sumWellRates(const Group2& group, const Schedule& schedule, const WellStateFullyImplicitBlackoil& wellState, const int reportStepIdx, const int phasePos, const bool injector, double& rate) {
+
+        for (const std::string& groupName : group.groups()) {
+            const Group2& groupTmp = schedule.getGroup2(groupName, reportStepIdx);
+            sumWellRates(groupTmp, schedule, wellState, reportStepIdx, phasePos, injector, rate);
+            rate *= groupTmp.getGroupEfficiencyFactor();
+        }
+        const auto& end = wellState.wellMap().end();
+        for (const std::string& wellName : group.wells()) {
+            const auto& it = wellState.wellMap().find( wellName );
+            if (it == end)  // the well is not found
```

Comment:
General nitpick: braces please... also some other minor things that could/should be handled by clang-format, could we agree on using it for new files such as this?